### PR TITLE
MAISTRA-2227 ignore empty yaml bytes

### PR DESCRIPTION
**Pull request by rcernich**
Signed-off-by: rcernich <rcernich@redhat.com>

### CODE REVIEW

**Review by knrc — APPROVED** [review]

File: pkg/controller/common/helm/manifestprocessing.go [hunk 1]
```diff
@@ -73,10 +73,14 @@ func (p *ManifestProcessor) ProcessManifest(ctx context.Context, man manifest.Ma
 			allErrors = append(allErrors, err)
 			continue
 		}
+		if len(rawJSON) == 0 || string(rawJSON) == "{}" || string(rawJSON) == "null" {
+			// MAISTRA-2227 ignore empty objects.  this could happen if charts have in empty blocks, e.g. superfluous "---"
+			continue
+		}
```
Comment (jwendell) [inline pkg/controller/common/helm/manifestprocessing.go:79]: Can we have a test for this?

File: pkg/controller/common/helm/manifestprocessing.go [hunk 2]
```diff
@@ -73,10 +73,14 @@ func (p *ManifestProcessor) ProcessManifest(ctx context.Context, man manifest.Ma
...
 		obj := &unstructured.Unstructured{}
 		_, _, err = unstructured.UnstructuredJSONScheme.Decode(rawJSON, nil, obj)
 		if err != nil {
-			log.Error(err, "unable to decode object into Unstructured")
+			log.Error(err, fmt.Sprintf("unable to decode object into Unstructured:\n%s\n", raw))
```
Comment (brian-avery) [inline pkg/controller/common/helm/manifestprocessing.go]: is there any chance we'd log something we don't intend to here?

Comment (brian-avery) [reply pkg/controller/common/helm/manifestprocessing.go]: I think it's ok. I think secrets and similar would be in other resources.

**Review by brian-avery — APPROVED** [review]

File: pkg/controller/common/helm/manifestprocessing.go (hunk 2 above)
Comment (rcernich) [reply pkg/controller/common/helm/manifestprocessing.go]: okay, made this more like a debug entry, leaving the error as it was

File: pkg/controller/common/helm/manifestprocessing.go (hunk 1 above)
Comment (rcernich) [reply pkg/controller/common/helm/manifestprocessing.go:79]: done

Comment (jwendell) [reply pkg/controller/common/helm/manifestprocessing.go:79]: I think the original error was not thrown in this function, was it? If my guess is true, then checking if err != nil in the test would not be enough, right?

**Comment by dgn** [comment]
Is this something that can happen in 2.0, too?

**Comment by rcernich** [comment]
If the charts have superfluous `---`, with no yaml between them and especially if there are comments those empty yaml blocks.

**Pull request merged**